nck/readers/mytarget_reader.py

```diff
@@ -0,0 +1,408 @@
+            raise ValueError("Either the start date or the end date is empty")
+        else:
+            return True
+
+    def __check_end_posterior_to_start(
+        self,
+        start_date: datetime,
+        end_date: datetime
+    ) -> bool:
+        if start_date > end_date:
+            raise ValueError(f"The start date {start_date} is posterior to end date {end_date}")
+        else:
+            return True
+
+    def __retrieve_and_set_token(self):
+        """In order to request the api, we need an active token. To do so we use the token which
+        was provided to get a new one which is going to be active for a day. Once done we set it
+        as an attribute.
+        """
+        parameters_refresh_token = self.__generate_params_dict('refresh_agency_token')
+        request_refresh_token = self.__create_request('refresh_agency_token', parameters_refresh_token)
+        refreshed_token = requests.post(**request_refresh_token).json()
+        self.set_agency_client_token(refreshed_token)
+
+    def __retrieve_all_data(
+        self) -> Tuple[
+            Dict[str, List[Dict[str, Any]]],
+            List[Dict[str, str]],
+            Dict[str, str],
+            Dict[str, str]]:
+        response_id = self.__get_all_results('get_campaign_ids')
+        response_name = self.__get_all_results('get_campaign_names')
+
+        names_dict = self.convert_list_dicts_to_dict(response_name, 'name')
+        ids_dict = self.convert_list_dicts_to_dict(response_id, 'campaign_id')
+
+        campaign_ids = [element['campaign_id'] for element in response_id]
+        rsp_banner = self.get_all_banners_all_camp(campaign_ids)
+
+        rsp_daily_stat = self.__get_response('get_banner_stats', banner_ids=rsp_banner.keys())
+        rsp_banner_names = self.get_banner_name_response('get_banner_names', list(rsp_banner))
+
+        return rsp_daily_stat, names_dict, ids_dict, rsp_banner_names
+
+    def __get_all_results(
+        self,
+        name_content: str,
+        offset=0,
+        campaign_id=0,
+        banner_ids=[]
+    ) -> List[Dict[str, str]]:
+        """Based on the __get_response function this function is incrementing through offsets according to the
+        number of elements given by the first response.
+
+        Args:
+            name_content (str): string representing key of parameters config dict
+            offset (int, optional): potential offset of the request. Defaults to 0.
+            campaign_id (int, optional): potential campaign id of the request. Defaults to 0.
+            banner_ids (list, optional): potential banner ids list of the request. Defaults to [].
+
+        Returns:
+            List[Dict[str, Any]]: list of dicts resulting from the requests we made to the api
+        """
+        first_elements = self.__get_response(name_content, campaign_id=campaign_id, banner_ids=banner_ids)
+        count = first_elements['count']
+        elements = [first_elements['items']]
+        if count > 20:
```

Comment:
We could have defined a constant at the top of the file or in the config and call it here. Thus, we would know the meaning of this number. The constant could have a little documentation on top with a link to the MyTarget doc if needed.